app/src/main/java/net/gsantner/markor/ui/hleditor/ReplacePatternGenerator.java

```diff
@@ -0,0 +1,33 @@
+package net.gsantner.markor.ui.hleditor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+public abstract class ReplacePatternGenerator {
+    protected List<TextActions.ReplacePattern> replaceNonSelectedPatternsWithSelectedOrReplaceWithAlternative(
+            final Pattern[] allSearchPatterns,
+            final Pattern selectedSearchPattern, final String selectedReplacement,
+            final String alternativeReplacement
+    ) {
+        List<TextActions.ReplacePattern> replacePatterns = new ArrayList<>();
+        for (final Pattern searchPattern : allSearchPatterns) {
+            String replacement;
+            if (searchPattern != selectedSearchPattern) {
```

Comment:
Note that the comparison here was for object identity. This code worked because the patterns passed into it were all  'const' `static final`. If the pattern is constructed this will not match correctly.

Reply:
Good point, it still worked because the passed patterns stayed unique, but this makes the function error-prone. Fixed it.